### ISSUE TITLE
Linting for `throw`

### ISSUE THREAD

**Issue by skynet-gh** [clj-kondo/clj-kondo#2172]
**Is your feature request related to a problem? Please describe.**
When calling `throw` incorrectly (wrong arity/types), `clj-kondo` does not issue any warnings. For example given
`src/test.clj`:
```clj
(ns test)

(defn example []
  (throw "err" {:data 1}))
```
Clojure errors with
```sh
$ clj -X test/example
Syntax error compiling throw at (test.clj:5:3).
Too many arguments to throw, throw expects a single Throwable instance
```
But `clj-kondo` does not warn
```sh
$ clj-kondo --lint src/
linting took 6ms, errors: 0, warnings: 0
```

**Describe the solution you'd like**
`clj-kondo` detects at least the arity if not the types for `throw` in Clojure (and maybe ClojureScript if possible).

**Describe alternatives you've considered**
It's a fairly minor annoyance granted, but nice to have for completeness I guess.
